### PR TITLE
add gallery component

### PR DESCRIPTION
**Pull request by himanshu9489**

### CODE REVIEW

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      margin: 22px 14px;
+    }
+    ${PREFIX} .previewCloseIcon.hideLeft, ${PREFIX} .previewCloseIcon.hideTop {
+      display: none
+    }
+    ${PREFIX} .galleryContainer {
+      display: flex;
+      flex: 1;
+      flex-direction: column;
+      align-items: center;
```
Comment (serialcoder13) [inline ui-app/client/src/components/Gallery/GalleryStyles.tsx:109]: Add as a theme property

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+      display: flex;
+      justify-content: flex-end;
+      gap: 20px;
+      cursor: pointer;
+      padding: 22px 12px;
+      background-color: rgb(24 24 27 / 50%);
+      z-index: 1;
+    }
+    ${PREFIX} .galleryToolbar .rightColumn i {
+      color: #FFFFFF;
```
Comment (serialcoder13) [inline ui-app/client/src/components/Gallery/GalleryStyles.tsx:140]: Add as a theme property

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    ${PREFIX} .thumbnailContainer.imageZoomed {
+      z-index: -1;
+    }
+    
+    ${PREFIX} .thumbnailContainer.thumbnailRight, ${PREFIX} .thumbnailContainer.thumbnailLeft {
+      flex-direction: column;
+      width: auto;
+      height: 80%;
+      margin: 0 10px;
+      max-height: none;
```
Comment (serialcoder13) [inline ui-app/client/src/components/Gallery/GalleryStyles.tsx:172]: Add as a theme property

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+      top: auto;
+      bottom: 0;
+      right: 0;
+      width:auto;
+    }
+    ${PREFIX} .button {
+      font-size: 50px;
+      cursor:pointer;
+      position: relative;
+      color: #FFFFFF;
```
Comment (serialcoder13) [inline ui-app/client/src/components/Gallery/GalleryStyles.tsx:269]: Add as a theme property

File: ui-app/client/src/components/Gallery/galleryProperties.ts [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+		COMPONENT_STYLE_GROUP_PROPERTIES.layout.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.position.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.spacing.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.typography.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.border.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.size.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.effects.type,
+		COMPONENT_STYLE_GROUP_PROPERTIES.background.type,
+	],
+};
```
Comment (serialcoder13) [inline ui-app/client/src/components/Gallery/galleryProperties.ts:364]: Add subcomponent styles for buttons, gallery images, thumbnail images, thumbnail containers, hover and other options

**Review by serialcoder13 — CHANGES_REQUESTED** [review]
Please make these changes

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx (hunk 1 above)
Comment (himanshu9489) [reply ui-app/client/src/components/Gallery/GalleryStyles.tsx:98]: don't need to add this one

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+      max-width: none;
+      transition: max-width .3s, max-height .3s, margin .3s;
+    }
+    ${PREFIX} .thumbnailContainer.imageZoomed {
+      z-index: -1;
+    }
+    
+    ${PREFIX} .thumbnailContainer.thumbnailRight, ${PREFIX} .thumbnailContainer.thumbnailLeft {
+      flex-direction: column;
+      width: auto;
```
[the remark this answers is not in the document]
Comment (himanshu9489) [reply ui-app/client/src/components/Gallery/GalleryStyles.tsx:151]: don't need to add this one

File: ui-app/client/src/components/Gallery/GalleryStyles.tsx (hunk 3 above)
Comment (himanshu9489) [reply ui-app/client/src/components/Gallery/GalleryStyles.tsx:152]: don't need to add this one

File: ui-app/client/src/components/Gallery/galleryProperties.ts (hunk 5 above)
Comment (himanshu9489) [reply ui-app/client/src/components/Gallery/galleryProperties.ts:364]: added

**Review by serialcoder13 — APPROVED** [review]
LGTM

**Pull request merged**